packages/realm-react/src/RealmProvider.tsx

```diff
@@ -78,11 +89,21 @@ export function createRealmProvider(
     // of the RealmProvider properties change.
     useEffect(() => {
       const combinedConfig = mergeRealmConfiguration(realmConfig, restProps);
-      if (!areConfigurationsIdentical(configuration.current, combinedConfig)) {
-        configuration.current = combinedConfig;
-        setConfigVersion((x) => x + 1);
+
+      // If there is a user in the current context and not one set by the props, then use the one from context
+      const combinedConfigWithUser =
+        user && !combinedConfig.sync?.user
+          ? mergeRealmConfiguration(combinedConfig, { sync: { user } })
+          : combinedConfig;
```

Comment:
Would this simplification be possible?

```suggestion
      const combinedConfigWithUser = mergeRealmConfiguration({ sync: { user } }, combinedConfig);
```

Reply:
With this suggestion, it is possible for a non-sync application to have {sync: {user: undefined}} included in the configuration.  But maybe there is a way to simplify this.

Reply:
```
 const combinedConfigWithUser =
        combinedConfig?.sync && user ? mergeRealmConfiguration({ sync: { user } }, combinedConfig) : combinedConfig;
```
This is a little simpler.

Reply:
> it is possible for a non-sync application to have {sync: {user: undefined}}

Yeah, I figured this could be an edge-case too and wondered if `mergeRealmConfiguration` implemented a deletion of the `sync` property if all members has `undefined` values or something.

Reply:
That's perhaps another possibility